langchain/llms/huggingface_pipeline.py

```diff
@@ -0,0 +1,104 @@
+"""Wrapper around HuggingFace Pipeline APIs."""
+from typing import Any, List, Mapping, Optional
+
+from pydantic import BaseModel, Extra
+
+import langchain
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+
+DEFAULT_MODEL_ID = "gpt2"
+DEFAULT_TASK = "text-generation"
+VALID_TASKS = ("text2text-generation", "text-generation")
+
+
+class HuggingFacePipeline(LLM, BaseModel):
+    """Wrapper around HuggingFace Pipeline API.
+
+    To use, you should have the ``transformers`` python package installed.
+
+    Only supports `text-generation` and `text2text-generation` for now.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms.huggingface_pipeline import HuggingFacePipeline
+            hf = HuggingFacePipeline(model_id="gpt2", task="text-generation")
+    """
+
+    pipeline: Any  #: :meta private:
+    model_id: str = DEFAULT_MODEL_ID
+    """Model name to use."""
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @classmethod
+    def from_model_id(
+        cls,
+        model_id: str,
+        task: str,
+        model_kwargs: Optional[dict] = None,
+        **kwargs: Any,
+    ) -> LLM:
```

Comment:
I could change this to `langchain.HuggingFacePipeline`, it wouldn't let me do just `HuggingFacePipeline`... Not sure why. Let me know if you want this to change

Reply:
no i think this is fine!